Source/WebCore/rendering/RenderLayerScrollableArea.h

```diff
@@ -225,6 +225,9 @@ class RenderLayerScrollableArea final : public ScrollableArea {
     void updateScrollbarsAfterStyleChange(const RenderStyle* oldStyle);
     void updateScrollbarsAfterLayout();
 
+    ScrollbarMode horizontalScrollbarMode() const final;
```

Comment:
I don't think `pointer-events:none` should affect the scrollbar mode (it doesn't affect how scrollbars render, only whether the user can interact).

Reply:
it was a bit fragile having completely different codepaths calling into [setScrollEnabled](https://github.com/WebKit/WebKit/blob/9ea306f30d73877433b5545e71f77fd1e64b1d9d/Source/WebKit/UIProcess/RemoteLayerTree/ios/ScrollingTreeScrollingNodeDelegateIOS.mm#L305) anyway.

Trying a simpler approach: a custom property on `WKChildScrollView`.